### PR TITLE
lightning: Fix empty region pre-check

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:
Lightning can not pass pre-check when user try to import a large of number tables but he does not disable `split-on-table` of TiDB configure.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Manual test (add detailed scripts or steps below)

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by glorv — APPROVED** [review]

File: br/pkg/lightning/restore/check_info.go [hunk 1]
```diff
@@ -263,7 +315,7 @@ func (rc *Controller) checkRegionDistribution(ctx context.Context) error {
 }
 
 // CheckClusterRegion checks cluster if there are too many empty regions or region distribution is unbalanced.
-func (rc *Controller) CheckClusterRegion(ctx context.Context) error {
+func (rc *Controller) CheckClusterRegion(ctx context.Context, controller *pdutil.PdController) error {
```
Comment (gozssky) [inline br/pkg/lightning/restore/check_info.go:318]: seems the new controller  argument is not used

File: br/pkg/lightning/restore/check_info.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"modernc.org/mathutil"
```
Comment (gozssky) [inline br/pkg/lightning/restore/check_info.go:22]: need move to next import group

**Review by gozssky — COMMENTED** [review]
rest LGTM

File: br/pkg/lightning/restore/check_info.go (hunk 1 above)
Comment (Little-Wallace) [reply br/pkg/lightning/restore/check_info.go:318]: Yes.... It is a bad code...

Comment (Little-Wallace) [reply br/pkg/lightning/restore/check_info.go]: Done

**Review by gozssky — APPROVED** [review]

**Comment by kennytm** [comment]
/run-check_dev_2

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/31616/pipeline

#15594

**Comment by niubell** [comment]
The issue https://github.com/pingcap/tidb/issues/27749 has affects-5.2 label, shouldn't we?

**Pull request merged**